ci/compare-images.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+#
+# This script serves to compare two docker images using skopeo tool. This gives
+# a brief information regarding the following image differences:
+#   - size
+#   - architecture
+#   - operating system
+#   - config
+#     - default user
+#     - exposed ports
+#     - environment variables
+#     - entrypoint
+#     - working directory
+#     - labels
+#   - Python packages
+#   - RPM packages
+#
+# It uses the skopeo TODO downloads images locally...
+#
+# Local execution: ./ci/compare-images.sh <image-1> <image-2>
+#   Note: <image-*> is in the format <repository@sha256:SHA>
+#
+# Example usage:
+#    ./ci/compare-images.sh quay.io/opendatahub/workbench-images@sha256:e92bf20e127e545bdf56887903dc72ad227082b8bc23f45ff4f0fc67e6430318 ghcr.io/jiridanek/notebooks/workbench-images:base-ubi9-python-3.9-jd_ubi_base_adedd4a943977ecdcb67bc6eb9eda572d10c3ddc
+
+shopt -s globstar
+
+
+function gather_metadata() {
+    local image="${1}"
+    local tmp_dir="${2}"
+
+    local ret_code=0
+
+    echo "Gathering the metadata for the image: '${image}'"
+
+    local image_sha
+    image_sha=$(echo "${image}" | cut -d ':' -f2)
+    echo "Image SHA: '${image_sha}'"
+
+    # Get image size
+    skopeo inspect --raw "docker://${image}" | jq '[ .layers[].size ] | add' > "${tmp_dir}/${image_sha}-size.txt"
+
+    # Get image metadata
+    skopeo inspect --config "docker://${image}" | jq -r '.architecture,.os,.config' > "${tmp_dir}/${image_sha}-metadata.txt"
+
+    # If we don't want to download the image, then we may consider to utilize the quay.io info:
+    # e.g.: https://quay.io/repository/opendatahub/workbench-images/manifest/sha256:f5a2c0666b5b03d68e6f9f2317b67f9bc5c3f4bd469bb7073dd144a33892f63a?tab=packages
+    # Disadvantage is that it takes some time this info is available on the quay
+
+
+    # Get image Python packages list
+    podman run --entrypoint /usr/bin/pip --rm -it "${image}" list > "${tmp_dir}/${image_sha}-global-pip.txt"
+    podman run --entrypoint /opt/app-root/bin/pip --rm -it "${image}" list > "${tmp_dir}/${image_sha}-local-pip.txt"
+
+    # Get image RPM packages list
+    podman run --entrypoint /usr/bin/rpm --rm -it "${image}" "-qa" > "${tmp_dir}/${image_sha}-rpms.txt"
+
+    echo "Metadata for image '${image}' gathered."
+}
+
+function compare_metadata() {
+    local tmp_dir="${1}"
+
+    echo "Let's compare the image metadata now:"
+
+    diff -y "${tmp_dir}"/*-size.txt
+    diff -y "${tmp_dir}"/*-metadata.txt
```

Comment:
This is essentially a json, or it can be printed as json, so the diff should use that

```
diff <(jq --sort-keys . A.json) <(jq --sort-keys . B.json)
```

some people on stackoverflow suggest `jd -set`, https://stackoverflow.com/questions/31930041/using-jq-or-alternative-command-line-tools-to-compare-json-files